src/movegen.cpp

```diff
@@ -156,10 +156,10 @@ namespace {
         {
             assert(rank_of(pos.ep_square()) == relative_rank(Us, RANK_6));
 
-            // An en passant capture can be an evasion only if the checking piece
-            // is the double pushed pawn and so is in the target. Otherwise this
-            // is a discovery check and we are forced to do otherwise.
-            if (Type == EVASIONS && !(target & (pos.ep_square() - Up)))
+            // An en passant capture cannot be an evasion if it is a discovery check
+            // and so the square the doubled-pushed pawn came from is in the target.
+            // Otherwise, the checking piece is the double-pushed pawn and it is pseudo-legal.
```

Comment:
Suggestion:

An en passant capture cannot be an evasion if there is a discovered check,
i.e. if the square the double pushed pawn came from is on an attacking line.
Otherwise, the checking piece is the double pushed pawn itself and the capture is pseudo-legal.

Reply:
Do I need to put _that is revealed by the double pawn push_? Because it already checked if there was a double pawn push and, for legal position, it must be a discovered check. Otherwise, black was already in check and it is a illegal position.

Reply:
Yeah it's redundant for legal reachable position. Oth, it hints at why the start square of double pushed pawn is checked.

Either way is fine for me though...